cmd/thanos/rule.go

```diff
@@ -521,31 +574,32 @@ func runRule(
 	)
 
 	// Start gRPC server.
-	{
-		tsdbStore := store.NewTSDBStore(logger, db, component.Rule, conf.lset)
+	tlsCfg, err := tls.NewServerConfig(log.With(logger, "protocol", "gRPC"), conf.grpc.tlsSrvCert, conf.grpc.tlsSrvKey, conf.grpc.tlsSrvClientCA)
+	if err != nil {
+		return errors.Wrap(err, "setup gRPC server")
+	}
 
-		tlsCfg, err := tls.NewServerConfig(log.With(logger, "protocol", "gRPC"), conf.grpc.tlsSrvCert, conf.grpc.tlsSrvKey, conf.grpc.tlsSrvClientCA)
-		if err != nil {
-			return errors.Wrap(err, "setup gRPC server")
-		}
+	options := []grpcserver.Option{
+		grpcserver.WithServer(thanosrules.RegisterRulesServer(ruleMgr)),
+		grpcserver.WithListen(conf.grpc.bindAddress),
+		grpcserver.WithGracePeriod(time.Duration(conf.grpc.gracePeriod)),
+		grpcserver.WithTLSConfig(tlsCfg),
+	}
+	if db != nil {
+		tsdbStore := store.NewTSDBStore(logger, db, component.Rule, conf.lset)
```

Comment:
In the future maybe we could also answer StoreAPI queries by using the data in the WAL? Perhaps worth adding a TODO here :smile:

Reply:
I am not sure about this. If the agent is able to query head data, then what's the difference between the agent and a prometheus with very low retention? I am not sure whether this is in their roadmap.

Reply:
Fair point :+1: